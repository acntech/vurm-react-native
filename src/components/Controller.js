import react from "react";
import { View, StyleSheet } from "react-native";
import ControllerButton from "./ControllerButton";
import { unicodeArrowToButtonLabelDirection } from "../utilities/conversion";

export default Controller = ({ setDirection }) => {
  const buttonRowUnicodeArrows = [["↑"], ["←", "→"], ["↓"]];
  return (
    <View style={styles.container}>
      <View style={styles.column}>
        {buttonRowUnicodeArrows.map((buttonUnicodeArrows) => (
          <View style={styles.row} key={buttonUnicodeArrows[0]}>
            {buttonUnicodeArrows.map((buttonUnicodeArrow) => (
              <ControllerButton
                key={buttonUnicodeArrow}
                label={buttonUnicodeArrow}
                onPress={() => {
                  setDirection(
                    unicodeArrowToButtonLabelDirection(buttonUnicodeArrow)
                  );
                }}
              ></ControllerButton>
            ))}
          </View>
        ))}
      </View>
    </View>
  );
};

const styles = new StyleSheet.create({
  container: {
    // backgroundColor: "aliceblue",
  },
  column: {
    flexDirection: "column",
    alignContent: "center",
  },
  row: {
    flexDirection: "row",
    alignContent: "center",
    justifyContent: "center",
  },
});

{
  /* {buttonLabels.map((buttonLabel) => (
          <TouchableOpacity
            key={buttonLabel}
            onPress={() => navigation.navigate(buttonLabel)}
            style={[styles.button]}
          >
            <Text style={[styles.buttonLabel]}>{buttonLabel}</Text>
          </TouchableOpacity>
        ))} */
}
