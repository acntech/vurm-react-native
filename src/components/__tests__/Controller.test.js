import React from "react";
import { render } from "@testing-library/react-native";
import Controller from "../Controller";

describe("<Controller />", () => {
  it("has 1 child", () => {
    // Arrange & Act
    const tree = render(<Controller />).toJSON();

    // Assert
    expect(tree.children.length).toBe(1);
  });

  it("renders correctly", () => {
    // Arrange & Act
    const tree = render(<Controller />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
