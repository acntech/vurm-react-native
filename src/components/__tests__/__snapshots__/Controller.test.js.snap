// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controller /> renders correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "coral",
              "borderRadius": 10000,
              "margin": 20,
              "minWidth": "50.3%",
              "opacity": 1,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 15,
                  "fontWeight": "900",
                  "textAlign": "center",
                },
              ]
            }
          >
            ↑
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "coral",
              "borderRadius": 10000,
              "margin": 20,
              "minWidth": "50.3%",
              "opacity": 1,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 15,
                  "fontWeight": "900",
                  "textAlign": "center",
                },
              ]
            }
          >
            ←
          </Text>
        </View>
      </View>
      <View
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "coral",
              "borderRadius": 10000,
              "margin": 20,
              "minWidth": "50.3%",
              "opacity": 1,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 15,
                  "fontWeight": "900",
                  "textAlign": "center",
                },
              ]
            }
          >
            →
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "coral",
              "borderRadius": 10000,
              "margin": 20,
              "minWidth": "50.3%",
              "opacity": 1,
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 15,
                  "fontWeight": "900",
                  "textAlign": "center",
                },
              ]
            }
          >
            ↓
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
